src/resolve/resolved-etc-hosts.c

```diff
@@ -22,6 +22,7 @@ static EtcHostsItemByAddress *etc_hosts_item_by_address_free(EtcHostsItemByAddre
         if (!item)
                 return NULL;
 
+        item->canon_name = NULL;
```

Comment:
Hmm, we don't zero other fields in `item`. So I don't think we should this one either.

Reply:
yes, please drop this